angular/index.d.ts

```diff
@@ -756,7 +756,7 @@ declare namespace angular {
          * @param comparator Function used to determine the relative order of value pairs.
          * @return An array containing the items from the specified collection, ordered by a comparator function based on the values computed using the expression predicate.
          */
-        <T>(array: T[], expression: string|((value: T) => any)|(((value: T) => any)|string)[], reverse?: boolean, comparator?: IFilterOrderByComparatorFunc): T[];
```

Comment:
Looks great.

It doesn't work for `orderByFilter(null, x => x)`, but I can't find what can be done about it.

To make the tests stricter, I would add type annotations:
```js
function testOrderByFilter() {
    const orderByFilter = angular.injector(['ng']).get<angular.IFilterOrderBy>('orderByFilter');
    const ordered: number[] = orderByFilter([1, 2, 3], x => x.toFixed()); // number[]
    const ordered1: null[] = orderByFilter([null, null], x => x); // null[]
    const orderedNull: null = orderByFilter(null); // null
    const orderedUndefined: undefined[] = orderByFilter([undefined, undefined], x => x); // undefined[]
}
```

P.S. trying to find a better solution, I've surprisingly discovered that it's valid TypeScript: `interface Foo { <T>(bar: T | T[]): typeof bar; }`. I mean the `typeof` part. Not sure what the use can be though.